jfxtras-agenda/src/main/java/jfxtras/internal/scene/control/skin/agenda/base24hour/AgendaSkinTimeScale24HourAbstract.java

```diff
@@ -238,7 +241,15 @@ public void refresh() {
 		refreshLocale();
 		setupAppointments();
 		nowUpdateRunnable.run(); 
-	}
+	}
+
+	/**
+	 * 
+	 */
+	@Override
+    public Pane getNodeForPopup(Appointment appointment) { return appointmentNodeMap.get(appointment); }
+    final private Map<Appointment, Pane> appointmentNodeMap = new WeakHashMap<>();
```

Comment:
I'm a bit worried about the hashcode method on the appointments and the hashmap being used here. What if one of the properties is changed after an appointment is put in this map?